### DISCUSSION TITLE
Restart error

### DISCUSSION THREAD

**Discussion by PEI0214** [idaholab/moose#28418] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I encountered the following error when using the restart:
~~~
Elem Information
   id()=338, unique_id()=38454, subdomain_id()=1, processor_id()=6
   type()=HEX8
   dim()=3
   n_nodes()=8
   mapping=LAGRANGE_MAP
    0  Node id()=709, processor_id()=0, Point=(x,y,z)=( 23.4213, -61.3682,  728.971)
    DoFs=(0/0/474) (0/1/475) (0/2/476) 
    1  Node id()=712, processor_id()=0, Point=(x,y,z)=( 23.4213, -61.1082,  728.587)
    DoFs=(0/0/483) (0/1/484) (0/2/485) 
    2  Node id()=712, processor_id()=0, Point=(x,y,z)=( 23.4213, -61.1082,  728.587)
    DoFs=(0/0/483) (0/1/484) (0/2/485) 
    3  Node id()=710, processor_id()=0, Point=(x,y,z)=(   23.43, -62.1282,  728.587)
    DoFs=(0/0/477) (0/1/478) (0/2/479) 
    4  Node id()=723, processor_id()=6, Point=(x,y,z)=( 20.2213,  -55.526,   732.86)
    DoFs=(0/0/50934) (0/1/50935) (0/2/50936) 
    5  Node id()=726, processor_id()=6, Point=(x,y,z)=( 20.2213,  -55.266,  732.476)
    DoFs=(0/0/50943) (0/1/50944) (0/2/50945) 
    6  Node id()=726, processor_id()=6, Point=(x,y,z)=( 20.2213,  -55.266,  732.476)
    DoFs=(0/0/50943) (0/1/50944) (0/2/50945) 
    7  Node id()=724, processor_id()=6, Point=(x,y,z)=(   20.23,  -56.286,  732.476)
    DoFs=(0/0/50937) (0/1/50938) (0/2/50939) 
   n_sides()=6
    neighbor(0)=331
    neighbor(1)=23716
    neighbor(2)=nullptr
    neighbor(3)=nullptr
    neighbor(4)=336
    neighbor(5)=351
   hmin()=0, hmax()=8.51517
   volume()=0.612496
   active()=1, ancestor()=0, subactive()=0, has_children()=0
   parent()=nullptr
   level()=0, p_level()=0
   refinement_flag()=DO_NOTHING
   p_refinement_flag()=DO_NOTHING
   DoFs=(1/0/126424) (1/1/126425) (1/2/126426) (1/3/126427) (1/4/126428) (1/5/126429) (1/6/126430) (1/7/126431) 
libMesh terminating:
ERROR: negative Jacobian -0 at point (x,y,z)=( 23.4213, -61.1082,  728.587) in element 338
[6] ../src/fe/fe_map.C, line 1138, compiled May 15 2024 at 11:58:00

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 6
~~~

The initial conditions for restarting are as follows:
~~~
[ICs]
  [disp_x]
    type = FunctionIC
    variable = disp_x
    block = '4    5    6'
    function = 0
  []
  [disp_y]
    type = FunctionIC
    variable = disp_y
    block = '4    5    6'
    function = 0
  []
  [disp_z]
    type = FunctionIC
    variable = disp_z
    block = '4    5    6'
    function = 0
  []
[]
~~~
There are some six-node cells in the grid, which are treated as follows:
![image](https://github.com/user-attachments/assets/7da6f653-05d2-4e14-b8a7-8520422cdbfe)

If no initial conditions are added, the system runs normally. How to solve this problem?

**Comment by GiudGiud** [comment]
Hello

We don't support 6 node elements. Do you mean you have 4 node elements with a non-conformality maybe? What does adding ``--mesh-info`` to the command line give in terms of statistics for the mesh?

**Reply by PEI0214** [reply]
Hello
Is 6-node unit not supported when the restart function is used? But when I do not set the initial condition in the restarted file, it works fine. Otherwise, an error will occur.
I tried a simple example and it worked. As shown in the image below, the restart is complete. 
![ca85705be2954766cbfee1935a879d4d](https://github.com/user-attachments/assets/1bf4abdd-219b-495d-be6a-09d6ea4f8863)
My grid is generated by an external grid generator. There are a lot of six-node cells. Do you have any suggestions to solve this problem?

**Reply by GiudGiud** [reply]
So you mean a 3D prism? 
That s supported

We don't support a 6 node 2D element

**Reply by PEI0214** [reply]
Yes, it's a 3D prism. Are the errors mentioned above caused by the grid?
I have another question, how do I merge the two result files from restart into one?
Thank you for your reply.

**Reply by GiudGiud** [reply]
> how do I merge the two result files from restart into one?

The only way to do that is to use either "one restart file and one SolutionUserObject" or "two SolutionsUserObjects". The SolutionUserObjects are paired with SolutionICs to initialize variables

> Are the errors mentioned above caused by the grid?

I dont understand what your 2D grid of numbers mean in the initial message
Where did you get this mesh with those prisms?
Is your mesh passing the mesh diagnostics? You can add one of these at the end to run them
https://mooseframework.inl.gov/source/meshgenerators/MeshDiagnosticsGenerator.html

**Reply by PEI0214** [reply]
Hello，
Since my grid is written as all 8 node cells, this causes some cells to have duplicate nodes, resulting in Jacobian 0. For example, the topology number of a 6-node prism is as follows.
![image](https://github.com/user-attachments/assets/9b92b7ec-e4ab-4c0f-bc5c-9344c05b358b)
When using reatart, the following error occurs. However, a direct calculation would not report the error. I need to use restart to implement some features, do you have any good suggestions? Can I skip the Jacobian check?
![2b5af1efea8a68f24c182de35aceb6fb](https://github.com/user-attachments/assets/2d34f5fb-6ffa-403a-ade5-ea14cad3337c)

**Reply by GiudGiud** [reply]
I don't understand this image with the numbers

Is the 6 node cell actually a 8 node cell with 2 nodes overlapping with two others?

Did you try the mesh diagnostics? With the MeshDiagnosticsGenerator

**Reply by GiudGiud** [reply]
and
- are you using adaptivity / mesh refinement in these simulations?
- which restart method are you using? (exodus or checkpoint?)
